### PR TITLE
Fix error in Enc-Dec Models

### PR DESCRIPTION
**Pull request by SSamDav**
Hi,

I think I fixed the errors in the Enc-Dec models, however, I have some doubts:

- Is the value corresponding to each time step? So you output the probabilities of the token and the value function?
- Why is everything in causal models shifted between logits and ids? Shouldn't the logits do a direct match with the ids?

### CODE REVIEW

**Comment by younesbelkada** [comment]
Closing in favor of https://github.com/lvwerra/trl/pull/93
I wanted to thank you again ! This PR helped us fixing the last issues with Enc-Dec support 💪 We made sure to add you as a co-author on https://github.com/lvwerra/trl/pull/93
